File: output/llama3.1-8b/avatar/Java/Python/atcoder_ABC123_C.py
import sys
from math import ceil

sc = input()
N = int(sc)
min_val = 0
for _ in range(5):
    if _ == 0:
        min_val = int(input())
    else:
        min_val = min(min_val, int(input()))

print(int(ceil(N / min_val)) + 4)

File: output/llama3.1-8b_IO_1/avatar/Python/Java/atcoder_ABC123_C.java
import java.util.Arrays;
import java.util.Scanner;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = Integer.parseInt(scanner.next());
        int[] A = new int[5];
        for (int i = 0; i < 5; i++) {
            A[i] = Integer.parseInt(scanner.next());
        }
        System.out.println((int) Math.ceil((double) N / Arrays.stream(A).min().getAsInt()) + 4);
    }
}
